### ISSUE TITLE
Add SolidColorBrush to App.xaml blows ActivationService

### ISSUE THREAD

**Issue by EdiWang** [Microsoft/WindowsTemplateStudio#968]
### Repro steps

1. Create a new project with Blank Page, MVVM Light, SettingsStorage, FirstRunDialog, WhatsNewDialog.
2. Add a SolidColorBrush or Color to App.xaml like this, e.g "BlowUpThings" :

```
<Application.Resources>
    <ResourceDictionary>
        <vms:ViewModelLocator x:Key="Locator" />
        <SolidColorBrush x:Key="BlowUpThings" Color="Red" />
        <ResourceDictionary.MergedDictionaries>
            <ResourceDictionary Source="/Styles/TextBlock.xaml"/>
        </ResourceDictionary.MergedDictionaries>
    </ResourceDictionary>
</Application.Resources>
```

3. Run the App

### Expected Behavior

ViewModelLocator should be initialized first to make sure "NavigationService.Frame.NavigationFailed" does not blow up.

### Actual Behavior

App blows up on start, exception happened in NavigationService.Frame.NavigationFailed, because the  constructor of ViewModelLocator is not invoked.

```
> System.InvalidOperationException occurred
>   HResult=0x80131509
>   Message=ServiceLocationProvider must be set.
>   Source=<Cannot evaluate the exception source>
>   StackTrace:
>    at Microsoft.Practices.ServiceLocation.ServiceLocator.get_Current()
>    at App1.Services.ActivationService.get_NavigationService() in C:\Workspace\App1\App1\Services\ActivationService.cs:line 24
>    at App1.Services.ActivationService.<ActivateAsync>d__6.MoveNext() in C:\Workspace\App1\App1\Services\ActivationService.cs:line 47
>    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
>    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
>    at System.Runtime.CompilerServices.TaskAwaiter.GetResult()
>    at App1.App.<OnLaunched>d__4.MoveNext() in C:\Workspace\App1\App1\App.xaml.cs:line 39
> 
```

### Note

A workaround is to put the SolidColorBrush into one of the ResourceDictionary.MergedDictionaries.

## System

* **VS Version:2017.15.3**
* **WTS Wizard Version:1.2.17198.1**
* **WTS Template Version:1.2.17202.1**
* **Windows Build:10.0.15063.540**

**Comment by EdiWang** [comment]
This seems only happen to Blank Page with MVVM Light. Navigation Pane template is fine.

**Comment by mrlacey** [comment]
Seems related to #414

**Comment by mvegaca** [comment]
Fixed on #1001

There was a problem because ActivationService was trying to get an instance of NavigationService before ViewModelLocator has registered the locator container and the NavigationService Instance.

**Comment by mvegaca** [comment]
Fix checked on
Wizard: 0.14.17243.65005
Templates: 0.14.17243.13
